lib/TTT/game.ex

```diff
@@ -1,38 +1,56 @@
 defmodule Game do
-  def init(token \\ "X") do
-    %State{player: token}
+  def init(player_token \\ "X") do
+    %State{player: player_token}
   end
 
+  def select_mode(mode) when mode == "1", do: :human_v_human
+  def select_mode(mode) when mode == "2", do: :human_v_computer
+
   def over(state) do
-    Output.display_board(state)
-    Output.end_of_game(state)
+    Messages.display_board(state)
+    Messages.end_of_game(state)
+  end
+
+  def play(state, deps, mode) when mode == :human_v_human do
+    deps.messages.display_board(state)
+
+    move(state, deps)
+    |> check_status(deps, :human_v_human)
   end
 
-  def play(state, deps) do
-    deps.output.display_board(state)
+  def play(state, deps, mode) when mode == :human_v_computer do
+    deps.messages.display_board(state)
 
-    new_state =
-      human_move(state, deps)
-      |> deps.output.display_board()
-      |> computer_move()
-      |> deps.output.display_message(:opponent_move)
+    move(state, deps)
+    |> computer_move()
+    |> check_status(deps, :human_v_computer)
+  end
 
+  def check_status(state, deps, mode) do
     cond do
-      Status.over(State.get_board(new_state)) == :game_in_progress ->
-        play(new_state, deps)
+      Status.over(State.get_board(state)) == :game_in_progress ->
+        play(state, deps, mode)
 
-      Status.over(State.get_board(new_state)) == :game_over ->
-        over(new_state)
+      Status.over(State.get_board(state)) == :game_over ->
+        over(state)
     end
   end
 
-  def human_move(state, deps) do
-    deps.input.retrieve(:choose)
-    |> deps.input.sanitized_move()
-    |> HumanPlayer.analyze(state, deps)
+  def move(state, deps) do
+    deps.io.retrieve(deps.messages.get_message(:choose))
+    |> deps.validation.sanitized_move()
+    |> deps.player.analyze(state, deps)
   end
```

Comment:
Might be clearer if this was renamed human_move or console_move, since there's a computer_move also